EmotiBitOscilloscope/src/ofApp.cpp

```diff
@@ -75,152 +75,7 @@ void ofApp::update() {
-				}
-				catch (...)
-				{
-					ofLogError("Failed to close pipe");
-				}
-				ofLog(OF_LOG_ERROR, "An exception occured while executing curl");
-				exceptionOccured = true;
-			}
-			try
-			{
-	#if defined (TARGET_OSX) || defined (TARGET_LINUX)
-				pclose(pipe);
-	#else
-				_pclose(pipe);
-	#endif
-			}
-			catch (...)
-			{
-				ofLogError("Failed to close pipe");
-			}
-		}
-		else
-		{
-			ofLog(OF_LOG_ERROR, "Failed to check for latest version. Failed to open pipe");
-			exceptionOccured = true;
-		}
-	}
-	catch (...)
-	{
-		ofLogError("Failed to open pipe");
-		exceptionOccured = true;
-	}
-	try 
-	{
-		if (!exceptionOccured & response != "")
-		{
-			ofxJSONElement jsonResponse;
-			if (jsonResponse.parse(response))
-			{
-				//ofLog(OF_LOG_NOTICE, jsonResponse.getRawString(true));  // uncomment to print curl output
-				std::string latestAvailableVersion = jsonResponse["tag_name"].asString();
-				ofLogNotice("Latest version") << latestAvailableVersion;
-				int swVerPrefixLoc = latestAvailableVersion.find(SOFTWARE_VERSION_PREFIX);
-				if (swVerPrefixLoc != std::string::npos)
-				{
-					latestAvailableVersion.erase(swVerPrefixLoc, 1);  // remove leading version char "v"
-				}
-				// compare with ofxEmotiBitVersion
-				std::vector<std::string> latestVersionSplit = ofSplitString(latestAvailableVersion, ".");
-				std::vector<std::string> currentVersionSplit = ofSplitString(ofxEmotiBitVersion, ".");
-				int versionLength = latestVersionSplit.size() < currentVersionSplit.size() ? latestVersionSplit.size() : currentVersionSplit.size();
-				if (versionLength)
-				{
-					for (int i = 0; i < versionLength; i++)
-					{
-						int latest = ofToInt(latestVersionSplit.at(i));
-						int current = ofToInt(currentVersionSplit.at(i));
-						if (latest == current)
-						{
-							// need to check minor version
-							continue;
-						}
-						else if (latest < current)
-						{
-							// current version higher than latest available
-							break;
-						}
-						else // latest > current
-						{
-							// new version available
-							newVersionAvailable = true;
-							break;
-						}
-					}
-				}
-				else
-				{
-					ofLogError("Failed to parse version string");
-				}
-			}
-			else
-			{
-				ofLog(OF_LOG_ERROR, "unexpected curl response");
-			}
-			// If newer version available, display alert message
-			if (newVersionAvailable)
-			{
-				// create alert dialog box
-				ofSystemAlertDialog("A new version of EmotiBit Software is available!");
-				// open browser to latest version
-				try
-				{
-#ifdef TARGET_WIN32
-					std::string command = "start " + latestReleaseUrl;
-					system(command.c_str());
-#else
-					std::string command = "open " + latestReleaseUrl;
-					system(command.c_str());
-#endif
-				}
-				catch (...)
-				{
-					ofLogError("Failed to open browser");
-				}
-			}
-		}
-	}
-	catch (...)
-	{
-		ofLogError("An exception occured while checking latest version of EmotiBit software");
-	}
-}
+
```

Comment:
This function has been moved to its own file in `ofxEmotiBit/src`.
Idea being, we can re-use it in other softwares (parser)